src/Data/Array/Accelerate/AST/Idx.hs

```diff
@@ -17,19 +25,19 @@ module Data.Array.Accelerate.AST.Idx
   where
```

Comment:
Maybe explicitly list exports here, and exclude `UnsafeIdxConstructor`? You don't seem to need it anywhere (yay!) and it closes a pretty gaping hole in the type-safety provided by having this in its own module.

Reply:
Yes, I'm glad we don't need it! I'm not sure yet whether I will want to use them in my AST rebuilding after fusion, but I'll add it and we can always export it again later